### ISSUE TITLE
A folder shared with two groups appears twice for an user in these two groups.

### ISSUE THREAD

**Issue by phlb** [owncloud/core#9724]
Hello,

When I try to share a folder with two groups, if a user is in these two groups, the user show the folder twice.

# Steps to reproduce

* create user 'test1'
* create user 'test2'
* create user 'test3'
* create a group 'group1' with the user test1 test2 test3
* create a group 'group2' with the user test1 test2
* with the user test1, create a folder "myfolder"
* with the user test1 share the folder "myfolder" with the group group1 without resharing neither modification allowed
* with the user test1 share the folder "myfolder" with the group group2 without resharing and with modification allowed

* login with the user test2

# Expected behavior

With the user test2, the folder "myfolder" should appear only once and modification should be allowed.

# Actual behaviour

With the user test2, the folder "myfolder" appears twice.

# Server configuration

**Operating system** : Debian Wheezy

**Database** : postgresql  9.1.13-0wheezy1

**php** : 5.4.4-14+deb7u9

**owncloud** : owncloud 7 RC3 / owncloud 6.0.3

**Updated from an older ownCloud or fresh install** :
Fresh install

**List of activated apps** :
Default installation

**Comment by lamuertepeluda** [comment]
👍  I'm experiencing the same behavior on 9.0.2

Folder `Projects` is shared as read-only with group A  
Folder `Projects/Project_Name` is shared as read/write/share with group B  

A user which belongs to both groups A and B, when accessing, will see the folder `Projects/Project_Name` twice in is home screen:

1. with read-only access in `Projects/Project_Name`
1. with read/write/share access as `Project_Name`

I would rater expect than he gets this `Projects/Project_Name` only, and he has group B permissions since they superseed group A. What do you think about it?

**Comment by PVince81** [comment]
@rullzer @schiesbn should we reopen or is there another ticket about "different permissions on group shares produces two recipient folders" ?

**Comment by rullzer** [comment]
@PVince81 not sure.
The main issue here is that grouping is not so trivial. And we miss a lot of info. We really should sit together and discuss this merging of shares. As it is not at all trivial..

**Comment by PVince81** [comment]
Ok, I'll reopen this ticket then.
Not sure why it was closed though, maybe autoclosed through a PR.
Last time we talked about it in some other ticket, I remember that we hadn't tackled the double folders due to different permissions.

**Comment by jammon88** [comment]
Hi there

Same issue with OC 9.0.3. In OC 8 we haven't that problem. Thanks for your check and fix.

**Comment by PVince81** [comment]
@jammon88 mind giving more details about your config and the way the shares are set up to confirm that it's the same issue ?

**Comment by jammon88** [comment]
Of course:

Server: Ubuntu 14.04.04
Database: mysql  Ver 14.14 Distrib 5.5.49
Webserver: Apache/2.4.7
Owncloud: 9.0.3

Share from a syncuser for LDAP synced groups. (two Groups)

Problem since update from OC 8.1.? to OC 9.0.3 with apt-get upgrade.

Config:

<?php
$CONFIG = array (
  'dbtype' => 'mysql',
  'version' => '9.0.3.2',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'installed' => true,
  'loglevel' => '2',
  'forcessl' => true,
  'ldapIgnoreNamingRules' => false,
  'theme' => '',
  'maintenance' => false,
  'share_folder' => '/Shared',
  'overwrite.cli.url' => '/owncloud',
  'trashbin_retention_obligation' => 'auto',
  'updatechecker' => false,
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'logtimezone' => 'Europe/Zurich',
  'log_type' => 'owncloud',
  'log_rotate_size' => 1048576,
); (rows with usernames, passwords, ids etc. I've deleted ;))

Do you need more infos? When yes, what?

**Comment by PVince81** [comment]
@jammon88 thanks, can you check the steps from https://github.com/owncloud/core/issues/9724#issue-38183197 whether it matches your sharing setup.

Or provide steps how to recreate the sharing situation you have on your server.
There are different versions of this issue (some involving different permissions), I just wanted to make sure which one it is. Thanks.

**Comment by jammon88** [comment]
We have different permissions, like in the post above. Group 1 has all right, Group 2 only read right. In the sharing folder named "shared" from the syncuser I've all right, in the sharing named "shared (1)" (looks like a linked folder) I've only read right with my personal user.

**Comment by PVince81** [comment]
Okay, thanks for the info.

**Comment by neoone256** [comment]
Hello, is it there any solution for this yet? I am having this problem on version 9.0.3

**Comment by norganna** [comment]
Confirmed again, would seem to have regressed in 9.0.3.

If I am in 2 groups to which a folder is shared, the folder appears twice.

**Comment by PVince81** [comment]
Yes, indeed. Closing as duplicate of https://github.com/owncloud/core/issues/24575

**Comment by tavinus** [comment]
Had 9.x.x? forgot (sry) and this was nnoying me. Just to let people know that 9.1.2 fixes this. I had to manually upgrade on 2 boxes though. Updater said it was up to date. 

The duplicated folders were removed from all accounts after installing it.

Side note: Manually upgrading is actualy easier and faster. I think I will always go this route from now on. So, don`t be scared if you need to manually upgrade as well.

**Comment by N0rimaki** [comment]
Are you People sure this behaviour is fixed? 9.1.4(stable) VM Image
I have still this Issue. 
-main
   -sub1
   -sub2

2 Groups 
group1
group2

-main (group1, group 2 all rights)
   -sub1 (grou1 one read rights)
   -sub2

Download on Windows 

-main
   -sub1 (group1 all rights)
   -sub2
-sub1 (group1 only read rights)
